commonui/src/main/java/com/hyperwallet/android/ui/common/view/error/DefaultErrorDialogFragment.java

```diff
@@ -104,18 +118,16 @@ public Dialog onCreateDialog(@Nullable Bundle state) {
                     EC_UNEXPECTED_EXCEPTION);
             errors.add(error);
         }
-
-        String message = ErrorUtils.getMessage(errors, getResources());
-
-        return buildDialog(errors.get(0).getCode(), message);
+        return errors;
     }
 
-    private AlertDialog buildDialog(@NonNull String errorCode, @NonNull String message) {
+    private AlertDialog buildDialog(@NonNull final Error error) {
+
         AlertDialog.Builder builder = new AlertDialog.Builder(
                 new ContextThemeWrapper(requireContext(), R.style.Theme_Hyperwallet_Alert));
-        builder.setMessage(message);
+        builder.setMessage(error.getMessage());
 
-        String errorType = ErrorTypes.getErrorType(errorCode);
+        String errorType = ErrorTypes.getErrorType(error.getCode());
         switch (errorType) {
```

Comment:
one more thing since, `errorType` is not used else where we can delete this and do this instead
```suggestion
        switch (ErrorTypes.getErrorType(error.getCode())) {
```

Reply:
done.